### PR TITLE
api & ui. collectd/rrd

### PR DESCRIPTION
**Pull request by andre8244**
Integrate api and ui for collectd/rrd queries

### CODE REVIEW

File: api/configuration/configuration.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ type Configuration struct {
 	ValuesPath             string          `json:"values_path"`
 	ViewsPath              string          `json:"views_path"`
 	PhonebookPath          string          `json:"phonebook_path"`
+	RrdRootPath            string          `json:"rrd_path"`
```
Comment (edospadoni) [inline api/configuration/configuration.go:57]: we can rename it to `RrdPath` ?

File: api/methods/queues.go [hunk 2]
```diff
@@ -80,17 +84,50 @@ func GetQueueReports(c *gin.Context) {
 
 	// data is cached, return immediately
 	if errCache == nil {
+		fmt.Println("got cached result for", graph, queryType) ////
```
Comment (edospadoni) [inline api/methods/queues.go:87]: we can remove it?

File: api/methods/queues.go [hunk 3]
```diff
@@ -100,32 +137,64 @@ func GetQueueReports(c *gin.Context) {
 	var queryString bytes.Buffer
 	errTpl := q.Execute(&queryString, &filter)
 	if errTpl != nil {
-		c.JSON(http.StatusBadRequest, gin.H{"message": "invalid query template compiling", "status": errTpl.Error()})
-		return
+		return "", errors.Wrap(errTpl, "invalid query template compiling")
 	}
 
+	fmt.Println("executing sql query:")                                                             ////
```
Comment (edospadoni) [inline api/methods/queues.go:143]: we can remove this debug prints?

File: api/methods/rrd.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+			}
+			record = append(record, value)
+		}
+		row++
+		data = append(data, record)
+	}
+	return data, nil
+}
+
+func ParseRrdResults(rrdData [][]interface{}) (string, error) {
```
Comment (edospadoni) [inline api/methods/rrd.go:123]: ParseRrdResults we can move it to `utils.go` with `ParseSqlResults`?

File: ui/src/views/QueueView.vue [hunk 5]
```diff
@@ -108,7 +108,26 @@ export default {
     retrieveQueryTree() {
       this.getQueryTree(
         (success) => {
-          this.queryTree = success.body.query_tree;
+          console.log("success.body.query_tree", success.body.query_tree); ////
```
Comment (edospadoni) [inline ui/src/views/QueueView.vue:111]: we can remove all `////` lines?

File: api/configuration/configuration.go (hunk 1 above)
Comment (andre8244) [reply api/configuration/configuration.go:57]: I chose `RrdRootPath` to not confuse it with `rrdFilePath` variable [used in rrd.go](https://github.com/nethesis/nethvoice-report/pull/14/files/37ce396bd824807de39a400e9ed77e828a7e103c#diff-d57c2910b18d24bc4274f59e29c20f7e9bfcd056d44d5916990c716704a62baeR30).
Do you think `RrdPath` is better anyway?

File: ui/src/views/QueueView.vue (hunk 5 above)
Comment (andre8244) [reply ui/src/views/QueueView.vue]: Removed. I left only some debug prints I think are still useful, we can delete them later

**Review by edospadoni — APPROVED** [review]

**Pull request merged**